capellambse/loader/core.py

```diff
@@ -692,6 +726,13 @@ def find_by_xsi_type(
             A list of XML elements to use as roots for the query.
             Defaults to all tree roots.
         """
+        import warnings
+
+        warnings.warn(
+            "find_by_xsi_type is deprecated, use `iterall_xt` instead",
+            DeprecationWarning,
+        )
```

Comment:
Do we need all the hassle with deprecating features of the low-level API? I mean with this PR you kind of make it public, such that we can expect usage of it too. But till then, I'd say we can change things as we go.

Reply:
I primarily threw in the warning there because I was concentrating on the docstrings at the time and didn't want to get distracted. :) Otherwise I probably would have changed it to defer to `iterall_xt` as well.

> with this PR you kind of make it public

It was already part of the public API before (it even shows up in the auto-generated APIDOC), it was just lacking an explanation of the broader concept on how things work together. So I would stick to standard procedures here too.